File: 0169/main.py
class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
# method(1) 利用Counter，值最大的那个
        from collections import Counter
        length = len(nums)//2 if(len(nums)%2==0) else len(nums)//2+1
        for i, var in Counter(nums).items():
            if(var>=length): return i

# method(2) 排序之后，中间那个值肯定是众数
#        nums.sort()
#        return nums[len(nums)//2]

File: 0169/main.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count=1, major;

        major = nums[0];
        for(int i=1;i<nums.size();i++) {
            if(nums[i]==major) {
                count++;
            } else {
                count--;
                if(count==0) major = nums[i+1];
            }
        }
        return major;
    }
};